### PR TITLE
1.4-M2 & HMPP fixes

### PR DESCRIPTION
**Pull request by h0tk3y**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -224,7 +222,6 @@ sourceSets {
     jvmMain {
         dependencies {
             implementation kotlin('stdlib-jdk8')
```
Comment (sandwwraith) [inline README.md:224]: Can JVM-only users still use `kotlinx-serialization-runtime` artifact or they have to migrate to `kotlinx-serialization-runtime-jvm`?

File: gradle.properties [hunk 2]
```diff
@@ -5,20 +5,21 @@
...
 junit_version=4.12
 infra_version=0.1.0-dev-47
-bintray_version=1.8.4-jetbrains-5
-benchmarks_version=0.1.7-dev-14
 jackson_version=2.10.0.pr1
 dokka_version=0.10.1
 native.deploy=false
 
+kotlin.mpp.enableGranularSourceSetsMetadata=true
+kotlin.mpp.enableCompatibilityMetadataVariant=true
```
Comment (sandwwraith) [inline gradle.properties:21]: What does this flag do?

File: README.md (hunk 1 above)
Comment (sandwwraith) [reply README.md:224]: Ok, judging by `publishPlatformArtifactsInRootModule` the answer is yes

Comment (dsavvinov) [reply README.md:224]: Yes, exactly

File: gradle.properties (hunk 2 above)
Comment (h0tk3y) [reply gradle.properties:21]: `enableCompatibilityMetadataVariant` is a special mode for libraries published with HMPP that adds an old-style `*.kotlin_metadata` variant that can be consumed by non-HMPP users. It is only useful for migration period until we enable HMPP by default.

Comment (dsavvinov) [reply gradle.properties:21]: In HMPP we've changed binary format of artifact published from common code. Previously it was a `.jar`-file with `.kotlin_metadata` inside, now it's a `.klib` without IR (well, essentially it's still an archive with `.kotlin_metadata` inside, but with different extension and layout)

This means that non-HMPP clients won't be able to consume that artifact. At the moment,  HMPP is not enabled by default, hence we need to publish a compatibility variant with old-style metadata .jar for anyone who hasn't enabled HMPP manually. 

If we enable HMPP by default in 1.4, then, most probably, we'll be able to drop this legacy-variant.

**Comment by dsavvinov** [comment]
> And I assume you have already tested teamcity uploading to bintray?

Yes, we've deployed HMPP libs to bintray multiple times via teamcity during development/testing. If you have some specific case in mind, let us know, we'll re-check it

File: gradle/publish-mpp-root-module-in-platform.gradle [hunk 3]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2014-2020 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+
+/** Publish the platform JAR and POM so that consumers who depend on this module and can't read Gradle module
+ metadata can still get the platform artifact and transitive dependencies from the POM: */
+project.ext.publishPlatformArtifactsInRootModule = { platformPublication ->
```
Comment (qwwdfsad) [inline gradle/publish-mpp-root-module-in-platform.gradle:8]: Please leave a comment to https://youtrack.jetbrains.com/issue/KMM-237#focus=streamItem-27-4115233.0-0 here

Comment (dsavvinov) [reply gradle/publish-mpp-root-module-in-platform.gradle:11]: Done!

**Pull request merged**